src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,84 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> clazzInterfaceImplementations;
+    private static final Map<Class<?>, Object> clazzInstances = new HashMap<>();
+
+    static {
+        clazzInterfaceImplementations = Map.of(
+                ProductService.class, ProductServiceImpl.class,
+                ProductParser.class, ProductParserImpl.class,
+                FileReaderService.class, FileReaderServiceImpl.class
+        );
+    }
+
+    private Injector() {
+
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
```

Comment:
Before creating a new instance, it is recommended to check if the instance already exists in the `clazzInstances` map to avoid creating duplicate instances.